### ISSUE TITLE
`aws_cognito_user_pool_client` `name` isn't updated

### ISSUE THREAD

**Issue by tomelliff** [terraform-providers/terraform-provider-aws#7525]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

---

Instead it just shows a perpetual diff and never updates it. It looks like this isn't being checked for in the Update method:
https://github.com/terraform-providers/terraform-provider-aws/blob/ebd84798973319148b91b75c2da9f2d63a8f68a4/aws/resource_aws_cognito_user_pool_client.go#L264-L324

### Terraform Version

Terraform v0.11.10
+ provider.aws v1.57.99

Note that the 1.57.99 version is because I'm building from a [fork](https://github.com/terraform-providers/terraform-provider-aws/pull/6314) that is based on the 1.57 release.

### Affected Resource(s)

* aws_cognito_user_pool_client

### Terraform Configuration Files

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "pool"
}

resource "aws_cognito_user_pool_client" "client" {
  name = "client"

  user_pool_id = "${aws_cognito_user_pool.pool.id}"
}
```

Followed by:

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "pool"
}

resource "aws_cognito_user_pool_client" "client" {
  name = "updated"

  user_pool_id = "${aws_cognito_user_pool.pool.id}"
}
```

### Expected Behavior

User pool client name should be updated.

### Actual Behavior

The name doesn't change and a diff is still shown on the next plan.

### Steps to Reproduce

1. `terraform apply` first HCL block
2. `terraform apply` second HCL block

**Comment by MatthewY-Circle** [comment]
I'm encountering similar behavior trying to add an `email_configuration` with just a `reply_to_email_address` under it -- Terraform reports success but it doesn't actually effect the change (and there's still a diff on next plan).

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.21.0 of the Terraform AWS Provider, later this week. 👍 Thanks to @jorcau for the fix implementation.

**Comment by bflad** [comment]
This has been released in [version 2.21.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2210-july-25-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests, documentation updates, or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
